src/binary/raw_binary_reader.rs

```diff
@@ -1176,24 +1098,32 @@ where
         self.data_source
             .read_slice(number_of_bytes, &mut self.buffer, slice_processor)
     }
+
+    /// Constructs an IonError::IllegalOperation which explains that the reader was asked to
+    /// perform an action that is only allowed when it is positioned over the item type described
+    /// byt the parameter `expected`.
+    fn expected<E: Display>(&self, expected: E) -> IonError {
+        illegal_operation_raw(format!(
+            "type mismatch: expected a(n) {} but positioned over a(n) {}",
+            expected,
+            self.current()
+        ))
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use std::io;
 
-    use bigdecimal::BigDecimal;
```

Comment:
PR tour: the unit tests were updated to use the new `StreamReader` API, so lots of `Option` wrappers were removed. Tests related to `read_datetime` and `read_big_decimal` were also removed.